src/test/java/org/opensearch/security/test/helper/rest/RestHelper.java

```diff
@@ -195,46 +221,34 @@ protected final String getHttpServerUri() {
 		log.debug("Connect to {}", address);
 		return address;
 	}
-	
-	protected final CloseableHttpClient getHTTPClient() throws Exception {
 
-		final HttpClientBuilder hcb = HttpClients.custom();
+	protected final CloseableHttpAsyncClient getHTTPAsyncClient(int numOfRequests) throws Exception {
+
+		final HttpAsyncClientBuilder hcb = HttpAsyncClients.custom().setMaxConnPerRoute(numOfRequests);
 
 		if (sendHTTPClientCredentials) {
```

Comment:
move to `getClientCredentials()`

Reply:
@vrozov what do you mean by move to  `getClientCredentials()`?

Reply:
@vrozov 
> The same for other if condition.

What if condition are you referring to?

Reply:
Do you mean
```
if (sendHTTPClientCredentials) {
    hcb.setDefaultCredentialsProvider(getClientCredentials());
}
```
in `getHTTPClient()` ?